nixos/modules/installer/tools/nixos-generate-config.pl

```diff
@@ -595,41 +655,172 @@ sub multiLineList {
+  # or
+  # services.xserver.videoDrivers = [ "amdgpu" ];
+  # or
+  # services.xserver.videoDrivers = [ "amdgpu-pro" ];
+
+  # Enable support for 32-bit OpenGL libraries on x86_64 which is
+  # useful if you want to run 32-bit Windows games under Wine.
+  #
+  # hardware.opengl.driSupport32Bit = true;
+  # environment.systemPackages = with pkgs; [ wine winetricks ];
+
+  # Some of the drivers above might not work with the current Linux
+  # kernel, so you might end up overriding that too.
+  #
+  # boot.kernelPackages = pkgs.linuxPackages_4_9;
 
   # Define a user account. Don't forget to set a password with ‘passwd’.
-  # users.extraUsers.guest = {
+  #
+  # users.users.me = {
   #   isNormalUser = true;
   #   uid = 1000;
+  #   extraGroups = [ "input" "audio" "video" "cdrom" ]
+  #              ++ [ "wheel" ]; # for sudo
+  #   openssh.authorizedKeys.keyFiles = [ /etc/nixos/id_rsa.pub ];
   # };
 
-  # This value determines the NixOS release with which your system is to be
-  # compatible, in order to avoid breaking some software such as database
-  # servers. You should change this only after NixOS release notes say you
-  # should.
-  system.stateVersion = "${\(qw(@release@))}"; # Did you read the comment?
+  # Enable the KDE Desktop Environment.
+  #
+  # services.xserver.displayManager.sddm.enable = true;
+  # services.xserver.desktopManager.plasma5.enable = true;
+
+  # Or maybe you want to enable Xfce Desktop Environment and Slim
+  # display manager?
+  #
+  # services.xserver.displayManager.slim.enable = true;
+  # services.xserver.desktopManager.xfce.enable = true;
+
+  # Or maybe you just want to automatically login into your .xsession
+  # script? Then don't enable any desktopManagers and uncomment the
+  # following.
+  #
+  # services.xserver.displayManager.auto = {
+  #   enable = true;
+  #   user = "me";
+  # };
+
+  # If you're in a hurry, you may stop editing your configuration.nix now
+  # and completely ignore everything below.
+
+  # It's generally considered a good practice to run networking
+  # software at the very least in a separate user accounts (if not in a
+  # separate containers or a VMs). NixOS does this automatically for
+  # the system services. However, like most other distros, at the moment
+  # NixOS doesn't do anything about the networking software that you run
+  # from your normal user accounts. Which is why you might want to setup
+  # separate user accounts for such things yourself.
+  #
+  # users.users.network = {
+  #   isNormalUser = true;
+  #   uid = 1001;
+  #
+  #   # Like systemPackages, but for this user only.
+  #   packages = pkgs; [ firefox thunderbird qtox ];
+  #
+  #   # You probably want it to play sound via ALSA
+  #   extraGroups = [ "audio" ];
+  #   # but you probably don't want to give it direct DRI access
+  #   # since OpenGL video drivers are pretty buggy and giving DRI
+  #   # access to WebGL might not be the best of ideas.
+  #   #          ++ [ "video" ];
```

Comment:
This instruction should be absolutely unnecessary: logged-in users should automatically have access to sound and video devices via loginctl / udev rules. If not, that's a bug that should be fixed.

Reply:
And how _not_ to give DRI access to a user even when logged in?

Reply:
> should automatically have access to sound and video devices via loginctl / udev rules

So you have a machine with a bunch of ssh accounts shared by a whole organization. Everyone should have access to sound and video devices by default? What about the "input" group that allows you to emulate key presses on evdev devices? I absolutely disagree.

Reply:
@oxij in that case the loginctl/udev rules would definitely need adjusting. Only logging in on the console should give you these permissions.